src/op/builder/gelu.cpp

```diff
@@ -0,0 +1,167 @@
+        auto half     = m.add_literal({x_type, {0.5f}});
+        auto one      = m.add_literal({x_type, {1.0f}});
+        auto sqrt2    = m.add_literal({x_type, {static_cast<float>(M_SQRT2)}});
+        auto mul_half = insert_common_op(m, ins, "mul", x, half);
+        auto div      = insert_common_op(m, ins, "div", x, sqrt2);
+        auto erf      = m.insert_instruction(ins, migraphx::make_op("erf"), div);
+        auto add_one  = insert_common_op(m, ins, "add", erf, one);
+        return {insert_common_op(m, ins, "mul", mul_half, add_one)};
+    }
+};
+
+struct gelu_tanh : op_builder<gelu_tanh>
+{
+    bool fast = false;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.fast, "fast"));
+    }
+
+
+    std::vector<instruction_ref>
+    insert(module& m, instruction_ref ins, const std::vector<instruction_ref>& args) const
+    {
+        auto x      = args[0];
+        auto x_type = x->get_shape().type();
+
+        auto fit_const_val  = fast ? 0.035677 : 0.044715;
+        auto fit_const      = m.add_literal({x_type, {fit_const_val}});
+        auto sqrt_2_rpi_val = fast ? 0.797885 : sqrt(M_2_PI);
+        auto sqrt_2_rpi     = m.add_literal({x_type, {sqrt_2_rpi_val}});
+        auto one            = m.add_literal({x_type, {1.0f}});
+        auto half           = m.add_literal({x_type, {0.5f}});
+        auto three          = m.add_literal({x_type, {3.0f}});
+
+        // [0.044715|0.035677] * x^3
+        auto pow0 = insert_common_op(m, ins, "pow", x, three);
+        auto mul0 = insert_common_op(m, ins, "mul", pow0, fit_const);
+        instruction_ref tanh_in;
+        if(fast)
+        {
+            // approx = 0.797885 * x + 0.035677 * x^3
+            auto mul1 = insert_common_op(m, ins, "mul", sqrt_2_rpi, x);
+            tanh_in   = insert_common_op(m, ins, "add", mul0, mul1);
+        }
+        else
+        {
+            // approx = sqrt(2/pi) * (x + 0.044715 * x^3
+            auto add0 = insert_common_op(m, ins, "add", mul0, x);
+            tanh_in   = insert_common_op(m, ins, "mul", add0, sqrt_2_rpi);
+        }
+
+        // 0.5 * x * (1 + Tanh(approx))
+        auto tanh0 = insert_common_op(m, ins, "tanh", tanh_in);
+        auto add1  = insert_common_op(m, ins, "add", tanh0, one);
+        auto mul2  = insert_common_op(m, ins, "mul", x, half);
+        return {insert_common_op(m, ins, "mul", add1, mul2)};
+    }
+};
+
+struct gelu_split : op_builder<gelu_split>
+{
+    template <class Self, class F>
+    static auto reflect(Self&, F)
+    {
+        return pack();
+    }
+
+
+    std::vector<instruction_ref>
+    insert(module& m, instruction_ref ins, const std::vector<instruction_ref>& args) const
+    {
+        auto x = args[0];
+        size_t last_dim_size = x->get_shape().lens().back();
+        if (last_dim_size < 2 or last_dim_size % 2 != 0)
+            MIGRAPHX_THROW("PARSE_GELU: BiasSplitGelu must have even last dimension which is >= 2");
```

Comment:
The error message should refer to the builder and not the onnx operator: `MIGRAPHX_THROW("gelu_split: must have even last dimension which is >= 2")`